lib/Conch/Controller/Login.pm

```diff
@@ -213,10 +248,32 @@ sub session_login ($c) {
 	# clear out all expired session tokens
 	$c->db_user_session_tokens->expired->delete;
 
-	$user->update({ last_login => \'NOW()' });
+	if ($user->force_password_change) {
+		$c->log->info('user ' . $user->name . ' logging in with one-time insecure password');
+		$user->update({
+			last_login => \'NOW()',
+			password => $c->random_string,	# ensure password cannot be used again
+		});
+		# password must be reset within 10 minutes
+		$c->session(expires => time + 10 * 60);
+
+		# conch-ui will now redirect the user to the change password form;
+		# conch-shell will print a warning to the user to change his password now.
+		return $c->status(200, {
```

Comment:
Should we be returning a 200 OK when it's really not ok because they need to change their password? paging @perigrin

Reply:
303 with a Location header is probably better.

Reply:
It should be 2xx, because we *are* logging them in.. they just need to do something afterwards to keep using the system.

Reply:
Does a redirect make sense, when we don't have a page to send them to that gives them a form to fill out and submit?  Or would the -ui intercept the redirect and send them to the equivalent form on its end?

Reply:
But as an API consumer, the only way I can know that is by parsing a free form 'message' attribute on the response payload.

Reply:
yeah, a freeform message doesn't make sense, i agree.